### PR TITLE
Allow tf.contrib.tpu._prepare_validation_data to modify x,y

### LINKED ISSUE #25034: Keras_to_tpu_model: validation_split parameter does not remove validation samples from training set
**System information**
- Using unmodified Google Colab TPU runtime

== cat /etc/issue ===============================================
Linux 07e8e3e8ed41 4.14.79+ #1 SMP Wed Dec 19 21:19:13 PST 2018 x86_64 x86_64 x86_64 GNU/Linux
VERSION="18.04.1 LTS (Bionic Beaver)"
VERSION_ID="18.04"
VERSION_CODENAME=bionic

== are we in docker =============================================
Yes

== compiler =====================================================
c++ (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux 07e8e3e8ed41 4.14.79+ #1 SMP Wed Dec 19 21:19:13 PST 2018 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
mesh-tensorflow          0.0.5                
msgpack-numpy            0.4.3.2              
numpy                    1.14.6               
protobuf                 3.6.1                
tensorflow               1.12.0               
tensorflow-hub           0.2.0                
tensorflow-metadata      0.9.0                
tensorflow-probability   0.5.0                

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.12.0
tf.GIT_VERSION = v1.12.0-0-ga6d8ffae09
tf.COMPILER_VERSION = v1.12.0-0-ga6d8ffae09
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

== cuda libs  ===================================================
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart.so.9.2.148
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-9.2/doc/man/man7/libcudart.7
/usr/local/cuda-9.2/doc/man/man7/libcudart.so.7

**Describe the current behavior**
When using the `tf.contrib.tpu.keras_to_tpu_model` API for training models on the TPU, providing a `validation_split` parameter does not remove the validation samples from the training set.  This causes the reported validation loss/metrics to be artificially high since we are training on samples that should have been withheld.

**Describe the expected behavior**
Specifically, when calling the fit method with a `validation_split` parameter, a proportion of the samples equal to (count of samples) * (validation_split) should be removed from the training data.

**Code to reproduce the issue**
A Colab notebook that reproduces the issue is here:
https://colab.research.google.com/drive/13g_eEYBUGw9GR6XVjop4TOQEKEDCJQVG

Specifically, a dummy dataset with 256 samples with validation_split = 1/8 should result in 224 training samples and 32 validation samples, but currently shows 256 training samples and 32 validation samples.

**Other info / logs**
I am very grateful for to everyone that is working on this interface.  This makes the on-ramp to TPU training much more manageable.

### PR DESCRIPTION
**Pull request by mike-holcomb**
When the user passes a validation_split parameter, the samples allocated to the val_x, val_y populations should be removed from the training data set (x, y).  In the _prepare_validation_split function, the validation split parameter branch already creates the correctly truncated x, y but does not return those modified x, y values back.  As a result, the validation samples are in both the training data set and the validation data sets. The validation loss/metrics are giving artificially good results since the model is training also the samples in the validation set.

Fixes #25034

### CODE REVIEW

**Comment by mike-holcomb** [comment]
I signed it!

**Comment by mrry** [comment]
I'm not familiar with the code in tensorflow/contrib/cpu, so we'll need a reviewer from the TPU team to take a look at this. /cc @jhseu

**Review by rjpower — APPROVED** [review]
LGTM, thanks!
